src/main/java/org/openstreetmap/atlas/geography/atlas/AbstractAtlas.java

```diff
@@ -108,7 +108,12 @@ public static Atlas forOsmPbf(final Resource resource)
     @Override
     public Iterable<Area> areasCovering(final Location location)
     {
-        return this.getAreaSpatialIndex().get(location.bounds());
+        final Iterable<Area> areas = this.getAreaSpatialIndex().get(location.bounds());
+        return Iterables.filter(areas, area ->
```

Comment:
This works well, however how about this:
```
Iterables.stream(areas).filter(area -> {
    final Polygon areaPolygon = area.asPolygon();
    return areaPolygon.fullyGeometricallyEncloses(location);
});
```
You choose the one you find the most readable :)